Using Postman to explore Sidecar tutorial (#1668)
substrate-developer-hub/substrate-docs: content/md/en/docs/tutorials/integrate-with-tools/postman-sidecar.md

## Getting List of Endpoints

The first action we will perform using Postman is to send a `GET` request to Sidecar to return a list of all the active endpoints available for our API. This request requires only the `url` variable to be defined, and as stated earlier, unless you have changed the default value in your Sidecar initialization or are working outside of your development environment, the value has already been set for you.

From within the Postman desktop environment navigate to the `GET List of API Endpoints` request and select it.

![Navigating to the correct endpoint](/media/images/docs/tutorials/postman-sidecar/second_step.png)

Once you select that option, the main body of the client will change to show you the options available for this request. Since this action takes no parameters, and as such is a great first request to attempt, there is not much here to do. However, for future requests, you can experiment by providing different parameters and seeing the various responses provided by the API.

![Options available for the request](/media/images/docs/tutorials/postman-sidecar/third_step.png)

> Go ahead and click on the `Send` button. Once you do, you should very shortly after see the response section of the Postman client become populated with data. It will present the data by default in JSON format, but you can change that to XML, HTML, or plain text. It should start with the following fields:

Reviewer: ```suggestion
1. Click **Send**. 

   After you send the request, you should see the response section in Postman populated with data. 
   By default, the data is displayed in JSON format, but you can change response to XML, HTML, or plain text. 
   You should see information similar to the following fields:

   ```

```json
{
  "docs": "https://paritytech.github.io/substrate-api-sidecar/dist",
  "github": "https://github.com/paritytech/substrate-api-sidecar",
  "version": "14.1.2",
  "listen": "127.0.0.1:8080",
}
```

> At this point, you can save this response to an output file and use it to build against your own application needs. You can also save it as a Postman response example, which gets added as a sub-option underneath the `GET List of API Endpoints` selection in the left-hand menubar. This request is great for its simplicity, but what if you need to alter one of the values in the endpoint defined as a variable? That will be our next step.

Reviewer: ```suggestion
At this point, you can save this response to an output file and use it to build against your own application needs. 
You can also save it as a Postman response example added underneath the GET List of API Endpoints menu option. 
This request is great for its simplicity, but in many cases, you'll want to manipulate the values defined for variable to retrieve specific information. 
That's the next step inthis tutorial.
```

> ## Getting User Account Information

Reviewer: ```suggestion
## Get account information
```

> Now that you are familiar with making an uncomplicated request, let's perform another `GET` request this time modifying the value of one of the environment variables. In this request, you will request a specific account's balance on the chain. Using the [Substrate Node Template](https://github.com/substrate-developer-hub/substrate-node-template) as our working blockchain, we will query the account balance of the `Alice` user. To quickly get the account address for `Alice`, you can spin up the [Substrate Front End Template](https://github.com/substrate-developer-hub/substrate-front-end-template) and copy the account address seen in the browser.

Reviewer: ```suggestion
Now that you are familiar with making an uncomplicated request, let's perform another `GET` request but this time modifying the value of one of the environment variables. 
In this request, you'll request the balance for a specific account on the chain. 
Using the [substrate node template](https://github.com/substrate-developer-hub/substrate-node-template) as the working blockchain, this request will query the account balance of the `Alice` user. 
To get the account address for `Alice`, you can spin up the [Substrate Front End Template](https://github.com/substrate-developer-hub/substrate-front-end-template) and copy the account address seen in the browser.
```

![Copying Alice's Account Address](/media/images/docs/tutorials/postman-sidecar/fourth_step.png)

> Once you have the account address in your clipboard, return to the Postman client, select the `Environments` option from the far left-hand menubar, then select the `Dev` environment. This will bring up the list of environment variables and their values. For each variable you will see two fields: `initial value` and `current value`. The `initial value` is what the variable is loaded with when starting a fresh `Dev` environment, whereas the `current value` is for the current session. Additionally, `current value` is only saved locally and never sent back up to Postman to be shared with other team members using the same Postman API collection.

Reviewer: ```suggestion
After you have the account address in your clipboard, return to the Postman client and select **Environments**.
Select the **Dev** environment to display the list of environment variables and their values. 
For each variable, there's an initial value and a current value. 
The `initial value` is what the variable is loaded with when starting a fresh `Dev` environment.
The `current value` is the value for the current session. 
The `current value` is only saved locally and never sent to Postman or shared with other team members using the same Postman API collection.
```

![Account variable in Postman environment options](/media/images/docs/tutorials/postman-sidecar/fifth_step.png)

> As you can see from the screenshot above, the `current value` has been modified with an account address for `Alice`. In your own Postman client, paste the address you copied from the front end template into the `current value` text box. Once you have done so, navigate back to `Collections` from the far left-hand menubar and then back to the `Substrate API Sidecar` collection. From within the collection, choose the `GET Account Balance Info` inside the `Accounts Folder`.

Reviewer: ```suggestion
Paste the account address for `Alice` into the `current value` text box. 
Click **Collections** and open the **Substrate API Sidecar** collection. 
Select the **Accounts** folder, then select **GET Account Balance Info**.
```

![Account Balance Info option](/media/images/docs/tutorials/postman-sidecar/sixth_step.png)

You will see that the endpoint is defined with two variables:

```
{{url}}/accounts/{{account}}/balance-info
```

You have just defined the value for the `account` variable, so at this point you are ready to click the `Send` button and receive the response, which will look something like this:

```json
{
    "at": {
        "hash": "0x8e683afdc4d1730d31f8c9a62eacbd839a045f7555cb1bf9528b9136092f00d1",
        "height": "507"
    },
    "nonce": "0",
    "tokenSymbol": "UNIT",
    "free": "1152921504606846976",
    "reserved": "0",
    "miscFrozen": "0",
    "feeFrozen": "0",
    "locks": []
}
```

Similar to the previous example, you can save this as an example response within Postman or save it to an output file to assist in your test fixtures or other uses in your application building.

> ## What's Next?

Reviewer: ```suggestion
## What's next?
```

> You have accomplished quite a lot in this tutorial! You have successfully imported an Postman API collection, defined environment variables, made API requests in Postman with custom variable definitions, and examined the responses from the API. You now have a good working basis from which to continue to build your experience using Sidecar within Postman. There are more areas for you to explore on your own including writing tests inside Postman for debugging purposes, setting up Postman flows that allow you to connect a series of API REST requests, and just trying out all the other endpoints made available to you.

Reviewer: ```suggestion
In this tutorial, you learned how to:

- Import an Postman API collection.
- Define environment variables to use in a collection.
- Send API requests with custom variable definitions.
- Inspect and save the responses from the API. 

You now have a good working basis for using Sidecar within Postman and to continue building on this experience. There are more areas for you to explore on your own, including how to:
-
- Send requests using the other endpoints in the predefined collection.
- Write tests in Postman for debugging purposes
- Set up Postman flows that allow you to connect a series of API REST requests.
```
